src/docs/03-concepts/02-activities.md

```diff
@@ -77,11 +77,23 @@ To support such use cases, Cadence allows :activity: implementations that do not
 
 ## Local Activities
 
-Some of the :activity:activities: are very short lived and do not need the queing semantic, flow control, rate limiting and routing capabilities. For these Cadence supports so called _:local_activity:_ feature. :local_activity:Local_activities: are executed in the same :worker: process as the :workflow: that invoked them. Consider using :local_activity:local_activities: for functions that are:
+Some of the :activity:activities: are very short lived and do not need the queing semantic, flow control, rate limiting and routing capabilities. For these Cadence supports so called _:local_activity:_ feature. :local_activity:Local_activities: are executed in the same :worker: process as the :workflow: that invoked them. 
+
+What you will trade off by using local activities
+* Debuggability: There is no ActivityTaskScheduled and ActivityTaskStarted events. So you would not able to see the input. The worker is always the same as the decision worker.
+* More possibility of duplicated execution. Though regular activity could also execute multiple times, the technical guarantee doesn't change. But local activity has more chance of seeing this because the result is not recorded into history until DecisionTaskCompleted. This is also because when exeucting multiple local activities in a row, SDK(Java+Golang) would optmize recording in a way that only recording by interval. 
```

Comment:
```suggestion
* More possibility of duplicated execution. Though regular activity could also execute multiple times, the technical guarantee doesn't change. But local activity has more chance of seeing this because the result is not recorded into history until DecisionTaskCompleted. This is also because when executing multiple local activities in a row, SDK(Java+Golang) would optimize recording in a way that only recording by interval. 
```

also I'm not quite sure what the "optimize recording" part is claiming.